File: recipes/claude/01-setup-anthropic-sdks/test_anthropic.ts
import dotenv from "dotenv";
import Anthropic from "@anthropic-ai/sdk";

dotenv.config();

const apiKey = process.env.ANTHROPIC_API_KEY;
if (!apiKey) {
  console.error("FATAL: ANTHROPIC_API_KEY is missing from process.env");
  process.exit(1);
}

const anthropic = new Anthropic({
  apiKey: apiKey,
  maxRetries: 3,
  timeout: 10000,
});

async function runHealthCheck(): Promise<void> {
  try {
    const message = await anthropic.messages.create({
      model: "claude-3-5-sonnet-20241022",
      max_tokens: 64,
      messages: [{ role: "user", content: "Respond with the word PING." }],
    });

    const firstBlock = message.content[0];
    if (firstBlock && firstBlock.type === "text") {
      console.log(`TypeScript SDK Verified: ${firstBlock.text.trim()} (Model: ${message.model})`);
    } else {
      console.warn("Received non-text response block from API.");
    }
  } catch (error) {
    if (error instanceof Anthropic.APIError) {
      console.error(`Anthropic API Error [${error.status}]: ${error.message}`);
    } else {
      console.error("Unexpected runtime failure:", error);
    }
    process.exit(1);
  }
}

runHealthCheck();

File: recipes/claude/01-setup-anthropic-sdks/test_anthropic.py
import os
import sys
from dotenv import load_dotenv
from anthropic import Anthropic, APIConnectionError, APIStatusError

load_dotenv()

api_key = os.getenv("ANTHROPIC_API_KEY")
if not api_key:
    print("FATAL: ANTHROPIC_API_KEY is not set in the environment.", file=sys.stderr)
    sys.exit(1)

client = Anthropic(api_key=api_key)

try:
    response = client.messages.create(
        model="claude-3-5-sonnet-20241022",
        max_tokens=64,
        messages=[
            {"role": "user", "content": "Respond with the word PING."}
        ]
    )
    output_text = response.content[0].text.strip()
    print(f"Python SDK Verified: {output_text} (Model: {response.model})")
except APIConnectionError as e:
    print(f"Network failure reaching Anthropic API: {e.__cause__}", file=sys.stderr)
    sys.exit(1)
except APIStatusError as e:
    print(f"API HTTP status error [{e.status_code}]: {e.response}", file=sys.stderr)
    sys.exit(1)
